resources/js/components/ui/calendar/TinyCalendar.tsx

```diff
@@ -0,0 +1,216 @@
+import {currentDate, cx, getWeekInfo, htmlAttributes, useLangFunc} from "components/utils";
+import {DateTime} from "luxon";
+import {CgCalendarToday} from "solid-icons/cg";
+import {FaSolidArrowLeft, FaSolidArrowRight} from "solid-icons/fa";
+import {For, Index, Show, VoidComponent, createMemo, createSignal, mergeProps, splitProps} from "solid-js";
+import s from "./TinyCalendar.module.scss";
+
+export type DaysRange = [DateTime, DateTime];
+
+interface Props extends htmlAttributes.div {
+  locale: Intl.Locale;
+  /** The current selection visible in the tiny calendar. */
+  selection?: DaysRange;
+  /** The currently displayed month. */
+  month: DateTime;
+  minDaysOfPrevMonth?: number;
+  minDaysOfNextMonth?: number;
+  minWeeks?: number;
+  showWeekdayNames?: boolean;
+  holidays?: DateTime[];
+
+  /** Returns the range that should be marked as hovered when hovering a day. */
+  getHoverRange?: (hoveredDay: DateTime) => DaysRange | undefined;
+  /** The function called when month is changed in the view. Controller should change the month prop. */
+  setMonth: (month: DateTime) => void;
+  /** The function called when a day is clicked. It should typically set selection. */
+  onDayClick?: (day: DateTime) => void;
+}
+
+const DEFAULT_PROPS = {
+  minDaysOfPrevMonth: 2,
+  minDaysOfNextMonth: 3,
```

Comment:
jaka logika że dwie różne liczby? może jeden parametr albo w ogóle hardcoded? ktoś będzie ustawiał te propsy?

Reply:
Stała wysokość wydaje się ważnym feature'm, więc chyba najbliżej mi do skalibrowania tego do zawsze 6 wierszy i wytłumaczenia się z obliczeń w komentarzu (jakieś skrajne przypadki luty itp.). 

Stałe w propsach nie do końca mnie przekonują, bo jak widzę props to jak parametr do funkcji albo w API -- zakładam że będą podawane różne wartości. A w memo nie sądzę, że będziemy chcieli to różnie parametryzować.

Reply:
No dooooobra usunę propsy. Bo tu faktycznie nie tak bardzo wierzę że to będzie kiedyś ruszane. Ale nie tak bardzo czuję że to potrzebne :)